array_test.go

```diff
@@ -748,3 +748,41 @@ func TestArrayConvertContains(t *testing.T) {
 	value.chain.assertFailed(t)
 	value.chain.reset()
 }
+
+func TestArrayEvery(t *testing.T) {
```

Comment:
We also need a couple of tests for failed assertion:

* for one of the element in the middle, function fails; we should check that in this case the original array's chain is marked as failed; we should also check that the function was still invoked for every element, instead of stopping after first failure

* assertion fails for multiple or maybe all elements; we should check that the function was still invoked for every element, instead of stopping after first failure

Same for TestObjectEvery.